### PR TITLE
Ticket 2889. Adjust pageflag logic to apply appropriate pageflag based on redactions present

### PR DESCRIPTION
**Pull request by Aman-Hundal**
Added new pageflag logic to apply appropriate pageflag for a page based on the redactions (for deleting annotations) or current pageflags (for adding annotations) associated with the page. 
Example: For any deletion of annotations to a page -> after the annotation is removed,  if any remaining redaction on the corresponding page contains a withheld in full flag -> that pageflag takes precedence and is applied.
Example: For any addition of annotations to a page -> if the current pageflags associated with the page/document (DocumentPageFlag Table) contains a withheld in full  flag-> that pageflag takes precedence and is applied.

Previous logic before ticket = 
A) If a annotation is removed, no changes to pageflag is done UNLESS there is no remaining redactions on the page left (and then the pageflag will be removed). 
B) If a redaction is added, the page flag related to the annotation is applied to the page (and in turn the pageflag for a page is always the most recent annotation completed)

### CODE REVIEW

File: computingservices/ZippingServices/models/zipperproducermessage.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class zipperproducermessage(object):
-    def __init__(self,jobid,requestid,category,requestnumber,bcgovcode,createdby,ministryrequestid,filestozip,finaloutput,attributes,summarydocuments,redactionlayerid) -> None:
+    def __init__(self,jobid,requestid,category,requestnumber,bcgovcode,createdby,ministryrequestid,filestozip,finaloutput,attributes,summarydocuments=None,redactionlayerid=None) -> None:
         self.jobid = jobid
```
Comment (divyav-aot) [inline computingservices/ZippingServices/models/zipperproducermessage.py:3]: foldername is missing here. take from dev-marshal

File: api/reviewer_api/services/documentpageflagservice.py [hunk 2]
```diff
@@ -172,6 +173,26 @@ def updatepageflags(
...
+            DocumentPageflag.savepageflag(
+                requestid,
+                docflagobj["docid"],
+                docflagobj["version"],
+                json.dumps(docflagobj['pageflag']),
+                json.dumps(userinfo),
+                redactionlayerid,
+                json.dumps(docflagobj["attributes"]),
+            )
+
```
Comment (divyav-aot) [inline api/reviewer_api/services/documentpageflagservice.py:195]: try to do a bulk save instead of saving this in a loop.

Comment (Aman-Hundal) [reply api/reviewer_api/services/documentpageflagservice.py:195]: Done - thanks for the feedback!

File: computingservices/ZippingServices/models/zipperproducermessage.py (hunk 1 above)
Comment (Aman-Hundal) [reply computingservices/ZippingServices/models/zipperproducermessage.py]: Done !

**Review by aparna-aot — APPROVED** [review]

**Pull request merged**